cpp/src/cugraph.cu

```diff
@@ -375,23 +478,27 @@ gdf_error gdf_add_adj_list(gdf_graph *graph) {
     return GDF_SUCCESS;
 
   GDF_REQUIRE( graph->edgeList != nullptr , GDF_INVALID_API_CALL);
-  GDF_REQUIRE( graph->adjList == nullptr , GDF_INVALID_API_CALL);
+  GDF_REQUIRE( graph->edgeList->src_indices->dtype == GDF_INT32, GDF_UNSUPPORTED_DTYPE );
 
   if (graph->edgeList->edge_data != nullptr) {
     switch (graph->edgeList->edge_data->dtype) {
-      case GDF_FLOAT32:   return gdf_add_adj_list_impl<float>(graph);
-      case GDF_FLOAT64:   return gdf_add_adj_list_impl<double>(graph);
+      case GDF_FLOAT32:   return gdf_add_adj_list_impl<int, float>(graph);
```

Comment:
So we're mixing int32_t and int in this code, so, if "int" here intends "int32_t", shouldn't we better use int32_t here? (and many other places in this file, if the intention is really to use 32 bit integer (not just some integral number that will not become very big)).